### ISSUE TITLE
Enable the `improved_panic_error_reporting` feature in `sp-io` for our runtimes in a few releases

### ISSUE THREAD

**Issue by koute** [paritytech/polkadot#4885]
Substrate PR: https://github.com/paritytech/substrate/pull/10741
Background issue: https://github.com/paritytech/substrate/issues/10585

We should wait for a few releases until people have upgraded their clients (so that the new required host function is available) and then switch this feature on.

**Comment by dvdplm** [comment]
dq: what does "enable" actually mean here? Does this need a Polkadot PR?

**Comment by bkchr** [comment]
Yes, there is a feature that needs to be activated. I think it can be done now, but there is no real need to get it done for the next release.

**Comment by ordian** [comment]
Does fixing this also fix https://github.com/paritytech/substrate/issues/11828?

**Comment by koute** [comment]
Indeed; this was included in Polkadot 0.9.17, and the newest version that explicitly pops up in our [telemetry](https://telemetry.polkadot.io/#stats/0x91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3) is 0.9.24, so should be fine to enable it.

I'll push a PR tomorrow morning.
